File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:

        # Approach 1:
        # answer = list()

        # pos_stack = list()
        # neg_stack = list()

        # for elem in nums:
        #     if elem > 0:
        #         pos_stack.append(elem)
        #     else:
        #         neg_stack.append(elem)

        # for i in range(len(pos_stack)):
        #     answer.append(pos_stack[i])
        #     answer.append(neg_stack[i])

        # return answer
            
        # Approach 2:
        answer = [0] * len(nums)
        pos, neg = 0, 1

        for elem in nums:
            if elem > 0:
                answer[pos] = elem
                pos += 2
            else:
                answer[neg] = elem
                neg += 2

        return answer

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {

    // Approach 1
    // let answer = [];
    // let pos_stack = [];
    // let neg_stack = [];

    // for (let elem of nums){
    //     if (elem > 0){
    //         pos_stack.push(elem)
    //     }
    //     else{
    //         neg_stack.push(elem)
    //     }
    // }

    // let i = 0
    // while (i < pos_stack.length){
    //     answer.push(pos_stack[i])
    //     answer.push(neg_stack[i])
    //     i++
    // }

    // return answer

    // Approach 2
    let answer = [];
    let pos = 0
    let neg = 1

    for (let elem of nums){
        if (elem > 0){
            answer[pos] = elem
            pos = pos + 2
        }
        else{
            answer[neg] = elem
            neg = neg + 2
        }
    }

    return answer
};